lib/synapse/service_watcher/zookeeper.rb

```diff
@@ -388,6 +398,16 @@ def deserialize_service_instance(data)
       return host, port, name, weight, haproxy_server_options, labels
     end
 
+    def parse_child_name(child_name)
+      numeric_id = child_name[child_name.length-10..child_name.length]
```

Comment:
where does this magic number 10 come from? We should probs move it to some constant

Reply:
`When creating a znode you can also request that ZooKeeper append a monotonically increasing counter to the end of path. This counter is unique to the parent znode. The counter has a format of %010d -- that is 10 digits with 0 (zero) padding (the counter is formatted in this way to simplify sorting), i.e. "0000000001".`

https://zookeeper.apache.org/doc/current/zookeeperProgrammers.html#Sequence+Nodes+--+Unique+Naming

updated code with the above comments